README.rst

```diff
@@ -37,27 +36,66 @@ API
 
 The API is pretty simple, the following functions are provided in the ``DNSResolver`` class:
 
-* ``query(host, type)``: Do a DNS resolution of the given type for the given hostname. It returns an
-  instance of ``asyncio.Future``. The actual result of the DNS query is taken directly from pycares.
-  As of version 1.0.0 of aiodns (and pycares, for that matter) results are always namedtuple-like
-  objects with different attributes. Please check the `documentation
-  <http://pycares.readthedocs.org/latest/channel.html#pycares.Channel.query>`_
-  for the result fields.
+* ``query_dns(host, type)``: Do a DNS resolution of the given type for the given hostname. It returns an
+  instance of ``asyncio.Future``. The result is a ``pycares.DNSResult`` object with ``answer``,
+  ``authority``, and ``additional`` attributes containing lists of ``pycares.DNSRecord`` objects.
+  Each record has ``type``, ``ttl``, and ``data`` attributes. Check the `pycares documentation
+  <https://pycares.readthedocs.io/>`_ for details on the data attributes for each record type.
+* ``query(host, type)``: **Deprecated** - use ``query_dns()`` instead. This method returns results
+  in a legacy format compatible with aiodns 3.x for backward compatibility.
 * ``gethostbyname(host, socket_family)``: Do a DNS resolution for the given
   hostname and the desired type of address family (i.e. ``socket.AF_INET``).
-  While ``query()`` always performs a request to a DNS server,
+  While ``query_dns()`` always performs a request to a DNS server,
   ``gethostbyname()`` first looks into ``/etc/hosts`` and thus can resolve
-  local hostnames (such as ``localhost``).  Please check `the documentation
-  <http://pycares.readthedocs.io/latest/channel.html#pycares.Channel.gethostbyname>`_
-  for the result fields. The actual result of the call is a ``asyncio.Future``.
+  local hostnames (such as ``localhost``). The actual result of the call is a ``asyncio.Future``.
 * ``gethostbyaddr(name)``: Make a reverse lookup for an address.
+* ``getaddrinfo(host, family, port, proto, type, flags)``: Resolve a host and port into a list of
+  address info entries.
+* ``getnameinfo(sockaddr, flags)``: Resolve a socket address to a host and port.
 * ``cancel()``: Cancel all pending DNS queries. All futures will get ``DNSError`` exception set, with
   ``ARES_ECANCELLED`` errno.
 * ``close()``: Close the resolver. This releases all resources and cancels any pending queries. It must be called
   when the resolver is no longer needed (e.g., application shutdown). The resolver should only be closed from the
   event loop that created the resolver.
 
 
+Migrating from aiodns 3.x
+=========================
+
+aiodns 4.x introduces a new ``query_dns()`` method that returns native pycares 5.x result types.
```

Comment:
Perhaps a link to the pycares docs where the result types are described would help?